guix-science/packages/geant4.scm

```diff
@@ -0,0 +1,274 @@
+
+(define-public geant4-g4particlexs geant4-g4particlexs-3.1.1)
+
+
+(define-public geant4-g4abla-3.1 (geant4-datasets ; 0.1 MiB
+  "G4ABLA" "3.1"
+  "1v97q28g1xqwnav0lwzwk7hc3b87yrmbvkgadf4bkwcbnm9b163n"))
+
+(define-public geant4-g4abla geant4-g4abla-3.1)
+
+
+(define-public geant4-g4incl-1.0 (geant4-datasets ; 0.1 MiB
+  "G4INCL" "1.0"
+  "0z9nqk125vvf4f19lhgb37jy60jf9zrjqg5zbxbd1wz93a162qbi"))
+
+(define-public geant4-g4incl geant4-g4incl-1.0)
+
+
+(define-public geant4-g4pii-1.3 (geant4-datasets
+  "G4PII" "1.3"
+  "09p92rk1sj837m6n6yd9k9a8gkh6bby2bfn6k0f3ix3m4s8as9b2"))
+
+(define-public geant4-g4pii geant4-g4pii-1.3)
+
+
+(define-public geant4-g4ensdfstate-2.3 (geant4-datasets 
+  "G4ENSDFSTATE" "2.3"
+  "00wjir59rrrlk0a12vi8rsnhyya71rdi1kmark9sp487hbhcai4l"))
+
+(define-public geant4-g4ensdfstate geant4-g4ensdfstate-2.3)
+
+
+(define-public geant4-g4realsurface-2.2 (geant4-datasets ; optional
+  "G4RealSurface" "2.2"
+  "08382y1258ifs7nap6zaaazvabg72blr0dkqgwk32lrg07hdwm4r"))
+
+(define-public geant4-g4realsurface geant4-g4realsurface-2.2)
+
+
+(define-public geant4-g4tendl-1.3.2 (geant4-datasets ; optional
+  "G4TENDL" "1.3.2"
+  "0jdqmz3rz5a7yrq1mli6dj3bnmn73igf4fdxwfbl3rxywg38fa9v"))
+
+(define-public geant4-g4tendl geant4-g4tendl-1.3.2)
+
+
+
+;; start define geant4-releases using Qt5
+
+(define (geant4-releases pkgtarver dispver hash)
+  (package
+    (name "Geant4")
+    (version dispver)
+    (source (origin
+              (method url-fetch)
+              (uri (geant4-releases-urls pkgtarver))
+              (sha256 (base32 hash))))
+    (build-system cmake-build-system)
+    (arguments `(
+      #:tests? #f
+      #:configure-flags `(
+        "-DGEANT4_INSTALL_DATA=OFF" ; default OFF, enforce to ensure reproducity
+        "-DGEANT4_INSTALL_DATASETS_TENDL=OFF" ; default OFF
+        "-DGEANT4_BUILD_MULTITHREADED=ON" ; default OFF on non-unix-like system
+        "-DGEANT4_USE_INVENTOR_QT=OFF" ; default OFF, needs Coin3D+SoQt, Qt5:lib+headers
+        "-DGEANT4_USE_QT=ON" ; default OFF, needs Qt5, OpenGL
+        "-DGEANT4_USE_SYSTEM_CLHEP=OFF"
+        "-DGEANT4_USE_SYSTEM_EXPAT=OFF" ; default ON, see 'expat' below in (inputs) for reason
+        "-DGEANT4_USE_SYSTEM_ZLIB=OFF"
+        "-DGEANT4_USE_TBB=ON" ; needs TBB
+        "-DGEANT4_USE_OPENGL_X11=ON" ; default OFF
+        "-DGEANT4_USE_OPENGL_WIN32=OFF" ; default OFF
+        "-DGEANT4_INSTALL_PACKAGE_CACHE=ON" ; default ON, may be set to OFF when packaging
+        ;;"-DGEANT4_INSTALL_DATADIR=share" ; default CMAKE_INSTALL_DATAROOTDIR 
+        ;;"-DCMAKE_INSTALL_DATAROOTDIR=share" ; default share
+        ;; need data under /gnu/store/xxx...xxx-geant4-10.07.p01/share/Geant4-10.7.1/data
+        ) ; end of configure-flags
+      #:phases
+      (modify-phases %standard-phases
+          (add-before 'configure 'copy-datasets
+          (lambda* (#:key inputs outputs (name ,name) (version ,version) #:allow-other-keys) 
+            (let* (
```

Comment:
Is it possible to symlink these? Or does the software support a variable that points to one/multiple data directories?